src/python/dxpy/bindings/dxdataobject_functions.py

```diff
@@ -34,26 +34,28 @@
 from ..exceptions import DXError
 from ..compat import basestring
 
-def dxlink(object_id, project_id=None):
+def dxlink(object_id, project_id=None, field=None):
     '''
     :param object_id: Object ID or the object handler itself
     :type object_id: string or :class:`~dxpy.bindings.DXDataObject`
     :param project_id: A project ID, if creating a cross-project DXLink
     :type project_id: string
+    :param field: A field name, if creating a job-based object reference
+    :type field: string
 
     Creates a DXLink (a dict formatted as a symbolic DNAnexus object
     reference) to the specified object.  Returns *object_id* if it
     appears to be a DXLink already.
     '''
     if isinstance(object_id, DXDataObject):
         object_id = object_id.get_id()
```

Comment:
This part is confusing, the rest of the method is fairly readable. Why is dxlink returning a different type based on the content? It would be much more readable if dxlink always returned a dict (for example). 

Ways to address this are:
1) Add some documentation, explaining what is going on, and why. 
2) Splitting the code between the case where a bare bones object_id is returned, and where a dict is returned.